front/src/modules/simulationResult/components/SpaceTimeChart/WaypointsModal.tsx

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+
+import { Button, Checkbox } from '@osrd-project/ui-core';
+import { Alert } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { omit } from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+
+import type { PathProperties, TrainScheduleBase } from 'common/api/osrdEditoastApi';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+import useModalFocusTrap from 'utils/hooks/useModalFocusTrap';
+import { mmToKm } from 'utils/physics';
+
+type WaypointsModalProps = {
+  waypointsModalOpen: boolean;
+  setWaypointsModalOpen: (open: boolean) => void;
+  waypoints: NonNullable<PathProperties['operational_points']>;
+  filteredWaypoints: NonNullable<PathProperties['operational_points']>;
+  setFilteredWaypoints: (waypoints: NonNullable<PathProperties['operational_points']>) => void;
+  projectedTrainPath: TrainScheduleBase['path'];
+};
+
+const WaypointsModal = ({
```

Comment:
Do you think WaypointsPanel would be a better name ?
```suggestion
const WaypointsModal = ({
```

Reply:
Fine by me :)